src/app/parcels/fetchParcelTableData.ts

```diff
@@ -30,52 +35,158 @@ export const getCongestionChargeDetailsForParcels = async (
+    sortState: SortState<ParcelsTableRow>,
+    start?: number,
+    end?: number,
+    parcelIds?: string[]
+): Promise<ParcelsTableRow[]> => {
+    const processingData = await getParcelProcessingData(
+        supabase,
+        filters,
+        sortState,
+        start,
+        end,
+        parcelIds
+    );
+    const congestionCharge = await getCongestionChargeDetailsForParcels(processingData, supabase);
+    const formattedData = processingDataToParcelsTableData(processingData, congestionCharge);
+
+    return formattedData;
+};
+
+export const getParcelsCount = async (
+    supabase: Supabase,
+    filters: Filter<ParcelsTableRow, any>[]
+): Promise<number> => {
+    let query = supabase.from("parcels_plus").select("*", { count: "exact", head: true });
+
+    filters.forEach((filter) => {
+        if (filter.methodConfig.methodType === PaginationType.Server) {
+            query = filter.methodConfig.method(query, filter.state);
+        }
+    });
+
+    const { count, error } = await query;
+    if (error || count === null) {
+        console.error(error);
+        throw new DatabaseError("fetch", "parcels");
+    }
+    return count;
+};
+
+export const getParcelIds = async (
+    supabase: Supabase,
+    filters: Filter<ParcelsTableRow, any>[],
+    sortState: SortState<ParcelsTableRow>
+): Promise<string[]> => {
+    let query = supabase.from("parcels_plus").select("*");
+    filters.forEach((filter) => {
+        if (filter.methodConfig.methodType === PaginationType.Server) {
+            query = filter.methodConfig.method(query, filter.state);
+        }
+    });
+    if (sortState.sort && sortState.column.sortMethodConfig?.methodType === PaginationType.Server) {
+        query = sortState.column.sortMethodConfig.method(query, sortState.sortDirection);
+    } else {
+        query = query.order("packing_datetime", { ascending: false });
+    }
+    const { data, error } = await query;
+    if (error) {
+        console.error(error);
+        throw new DatabaseError("fetch", "parcels");
+    }
+    return data.map((parcel) => parcel.parcel_id ?? "") ?? [];
+};
 
+export const getAllValuesForKeys = async <Return>(
+    supabase: Supabase,
+    table: TableNames | ViewNames,
+    selectMethod: (
+        query: PostgrestQueryBuilder<Database["public"], any, any>
+    ) => PostgrestFilterBuilder<Database["public"], any, any>
+): Promise<Return> => {
+    const query = selectMethod(supabase.from(table));
+    const { data, error } = await query;
+    if (error) {
+        console.error(error);
+        throw new DatabaseError("fetch", table);
+    }
     return data ?? [];
 };
+
+export interface CollectionCentresOptions {
+    name: string;
+    acronym: string;
+}
+export interface StatusResponseRow {
+    event_name: string;
+}
+
+export interface RequestParams<Data> {
+    allFilters: Filter<Data, any>[];
+    sortState: SortState<Data>;
+    startPoint: number;
+    endPoint: number;
+}
+
+export const isFreshRequest = <Data>(
```

Comment:
is this function returning whether the two given requests are identical? What do you mean by isFreshRequest?

Reply:
good point, I guess this function doesn't care about the concept of fresh requests or whatever, it just takes in 2 request param objects and compares them, so I will rename it to areRequestsIdentical and rename the objects to requestParamsA and requestParamsB